cypress/pages/BounceRulesContainer/index.js

```diff
@@ -96,65 +96,72 @@ class BounceRulesPage extends Page {
   }
 
   deleteBounceRuleAPI(testRule) {
-    return new Promise(resolve => {
-      cy.task("getRules").then(res => {
-        if (!res.errno) {
+    return cy
+      .task("getRules")
+      .then(res => {
+        if (res) {
           for (let i = 0; i < res.length; i++) {
             if (_.isEqual(testRule, _.omit(res[i], "id"))) {
-              cy.task("deleteRule", res[i].id);
-              cy.log("DELETE SUCCESS");
+              return cy.task("deleteRule", res[i].id);
             }
           }
-        } else {
-          cy.log("DELETE ERROR");
+          return true;
         }
-        cy.reload().then(() => resolve());
+        return false;
+      })
+      .then(result => {
+        if (result) {
+          cy.log("Delete Successful");
+          cy.reload();
+        }
+        cy.log("Delete Unsuccessful");
       });
-    });
   }
 
   createBounceRuleAPI(testRule) {
-    return new Promise(resolve => {
-      cy.task("createRule", testRule).then(res => {
-        cy.reload().then(() => resolve(res.id));
-      });
-    });
+    return cy.task("createRule", testRule);
   }
 
   createBounceRuleUI(bounceRule) {
-    return new Promise(resolve => {
-      const {
-        priority,
-        bounce_action: bounceAction,
-        response_code: responseCode,
-        description,
-        enhanced_code: enhancedCode,
-        regex,
-      } = bounceRule;
-
-      this.createRuleButton.click();
-
-      if (priority) {
-        this.priority.type(priority);
-      }
-      if (bounceAction) {
-        this.bounceAction.type(bounceAction);
-      }
-      if (responseCode) {
-        this.responseCode.type(responseCode);
-      }
-      if (description) {
-        this.description.type(description);
-      }
-      if (enhancedCode) {
-        this.enhancedCode.type(enhancedCode);
-      }
-      if (regex) {
-        this.regex.type(regex);
-      }
-
-      resolve(this.submitButton.click());
-    });
+    const {
+      priority,
+      bounce_action: bounceAction,
+      response_code: responseCode,
+      description,
+      enhanced_code: enhancedCode,
+      regex,
+    } = bounceRule;
+
+    this.createRuleButton.click();
+
+    if (priority) {
+      this.priority.type(priority);
+    }
+    if (bounceAction) {
+      this.bounceAction.type(bounceAction);
+    }
+    if (responseCode) {
+      this.responseCode.type(responseCode);
+    }
+    if (description) {
+      this.description.type(description);
+    }
+    if (enhancedCode) {
+      this.enhancedCode.type(enhancedCode);
+    }
+    if (regex) {
+      this.regex.type(regex);
+    }
+
+    this.submitButton.click();
+    this.confirmModal.should("be.visible");
+    return this.confirmationSubmit.click();
+  }
+
+  deleteBounceRuleUI(id) {
+    cy.reload();
```

Comment:
Let's avoid using cy.reload() in the tasks and in here since to delete something as it shouldn't need to know to reload beforehand cause as a user you wouldn't reload first and then delete through the UI - it assumes too much that something happened before. Let's adjust our beforeEach instead to not have the open() before each test runs but rather you call Page.open() when you need to in each test and it'll make more sense and mimic what a user does better